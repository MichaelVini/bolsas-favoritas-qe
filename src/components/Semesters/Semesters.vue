<template>
    <div class="semester-list">
        <ul class="semester-list__items">
            <li class="semester-list__item" :class="{selected: allSemesters}"><a  @click.prevent="changeSemester($event.target.id)" id="1" class="semester-list__item__text">Todos os semestres</a></li>
            <li class="semester-list__item" :class="{selected: segundoSem2019}"><a @click.prevent="changeSemester($event.target.id)" id="2" class="semester-list__item__text">2° semestre de 2019</a></li>
            <li class="semester-list__item" :class="{selected: primeiroSem2020}"><a  @click.prevent="changeSemester($event.target.id)" id="3" class="semester-list__item__text">1° semestre de 2020</a></li>
        </ul>
    </div>
</template>

<script>
import { mapMutations, mapState } from 'vuex';
export default {
    data(){
        return {
            allSemesters: true,
            segundoSem2019: false,
            primeiroSem2020: false
        }
    },
    computed: {
        ...mapState(["semesterScholarships"])
    },
    methods: {
        ...mapMutations(["UPDATE_SEMESTERS_SCHOLARSHIPS"]),
         changeSemester(id){
            if(id == 1){
                this.allSemesters = true;
                this.segundoSem2019 = false;
                this.primeiroSem2020 = false;
            } else if (id == 2) {
                this.segundoSem2019 = true;
                this.allSemesters = false;
                this.primeiroSem2020 = false;
            } else {
                this.primeiroSem2020 = true;
                this.allSemesters = false;
                this.segundoSem2019 = false;
            }
            this.sendSemesters();            
        }, 
        sendSemesters(){
            this.UPDATE_SEMESTERS_SCHOLARSHIPS([this.allSemesters, this.segundoSem2019, this.primeiroSem2020])
        }
    },
}
</script>

<style src="./semesters.css" scoped>