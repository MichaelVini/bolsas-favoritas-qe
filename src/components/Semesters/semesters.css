.semester-list {
    margin: 60px auto;
    border-radius: 8px;
    height: 210px;
    width: 480px;
    color: var(--blue-secondary);
    font-weight: 900;
    font-size: 1.2em;
}

.semester-list__items {
    padding-inline-start: 0px;
}

.semester-list__item {
    padding: 20px;
    list-style: none;
    text-align: center;
    border: var(--blue-secondary) 2px solid;
    transition: 0.4s;
}

.semester-list__item:hover{
    background-color: var(--blue-secondary);
    color: white;
    cursor: pointer;
}
.semester-list__item.selected {
    background-color: var(--blue-secondary);
    color: white;
}

.semester-list__item:nth-child(1) {
    border-radius: 8px 8px 0px 0px;
}

.semester-list__item:nth-child(3) {
    border-radius: 0px 0px 8px 8px;
}

.semester-list__item__text {
    text-decoration: none;
    color: inherit;
}

@media (min-width: 1125px) {
   .semester-list {
        margin: 50px 0px -80px 0px;
        width: 80vw;
    }

    .semester-list__items {
        display: flex;
        justify-content: flex-end;
    }

    .semester-list__item:nth-child(1) {
        border-radius: 6px 0px 0px 6px;
    }
    
    .semester-list__item:nth-child(3) {
        border-radius: 0px 6px 6px 0px;
    }

    .semester-list__item {
        padding: 6px 26px;
        font-size: 0.9em;
    }
}
