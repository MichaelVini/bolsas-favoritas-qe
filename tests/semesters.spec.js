import { shallowMount } from '@vue/test-utils'
import Semesters from './../src/components/Semesters/Semesters.vue'

describe('Semesters.vue', () => {
    it('should render default state', () => {
        const wrapper = shallowMount(Semesters)
        
        expect(wrapper.html()).toMatchSnapshot()
    })
})
