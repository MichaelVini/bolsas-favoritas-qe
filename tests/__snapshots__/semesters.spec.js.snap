// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Semesters.vue should render default state 1`] = `
"<div class=\\"semester-list\\">
  <ul class=\\"semester-list__items\\">
    <li class=\\"semester-list__item selected\\"><a id=\\"1\\" class=\\"semester-list__item__text\\">Todos os semestres</a></li>
    <li class=\\"semester-list__item\\"><a id=\\"2\\" class=\\"semester-list__item__text\\">2° semestre de 2019</a></li>
    <li class=\\"semester-list__item\\"><a id=\\"3\\" class=\\"semester-list__item__text\\">1° semestre de 2020</a></li>
  </ul>
</div>"
`;
